### ISSUE TITLE
Can't complete on `<`

### ISSUE THREAD

**Issue by Compro-Prasad** [osv/company-web#20]
I am using spacemacs where I want to complete when I type the `<` character. But it doesn't complete. The `company-minimum-prefix-length` for me is `1`. So, I have to type an extra alphabet after `<` to get completion. How do I resolve this?
See: https://github.com/syl20bnr/spacemacs/issues/8222

**Comment by Compro-Prasad** [comment]
It might be unrelated to react layer which I don't use. But I tested this on an HTML file.

**Comment by osv** [comment]
I use spacemacs too
But my version is outdatted: 4bb4cb46  2017-11-01
For web-mode company-minimum-prefix-length is 0 for me. Maybe for you is 1 because of some perfomance issue. Try to set to 0 and check how perfomance decrased.

Few month ago there was performance issue for company-web and that is why they might decide to increase from 0 to 1 :). I'm closing this

**Comment by osv** [comment]
Ah, you are about react + company-web... I reopen, newer tried to work with react + company-web, need to check

**Comment by Compro-Prasad** [comment]
No, spacemacs has the value 0 but I manually set `company-minimum-prefix-length` to 1.

**Comment by Compro-Prasad** [comment]
It happens in a normal HTML file too.

**Comment by osv** [comment]
![autocomplete](https://user-images.githubusercontent.com/213405/41723957-1a5bd1e8-7575-11e8-9bbf-cb159857bf63.jpg)

What exactly happens? you want autocomplete after `<` ? Than set company-minimum-prefix-length to 0

**Comment by Compro-Prasad** [comment]
But I don't want to complete after `>`. Setting it to 0 means it will give me completion even after `>`.

**Comment by Compro-Prasad** [comment]
`company-clang` can complete on `::`, `->` insertions irrespective of `company-minimum-prefix-length`. Why can't this package do it?

**Comment by osv** [comment]
@Compro-Prasad Ah, reason why autocomplete in any place even after `>` is because web-mode is designed for **emmet** mode too!. 

In case if you are ok to work without emmet support in react layer we can update next function and add some switcher for disable **emmet**. 

```lisp
(defun company-web-html-emmet-grab ()
  (let* ((limit (company-web-backward-min-tag-bound))
         (bound (save-excursion
                  (if (re-search-backward ">" limit t)
                      (point)
                    limit))))
    (and company-web-html-emmet-enable
         (bound-and-true-p emmet-mode)
         (or
          (company-grab company-web-html-emmet-tag-regexp 1 bound)
          (company-grab company-web-html-emmet-class-regexp 2 bound)
          (company-grab company-web-html-emmet-id-regexp 2 bound)
          (company-grab company-web-html-emmet-attr-regexp 2 bound)
          (company-grab company-web-html-emmet-value-regexp 3 bound)))))
```

**Comment by osv** [comment]
I will add switcher soon. I think it should help you

**Comment by Compro-Prasad** [comment]
This is a bug IMO.
